File: python/examples/jobs/01_list_jobs_sdk.py
#!/usr/bin/env python3
"""
Example: List jobs using SDK

This example demonstrates:
- Listing all jobs with pagination
- Filtering jobs by status
- Accessing job metadata (timestamps, URLs, status)
"""

import asyncio
from crawl4ai_cloud import AsyncWebCrawler

# Configuration
API_KEY = "YOUR_API_KEY"  # Replace with your API key


async def main():
    async with AsyncWebCrawler(api_key=API_KEY) as crawler:
        # List all jobs (default: 20 per page)
        print("=== All Jobs (First 20) ===")
        jobs = await crawler.list_jobs(limit=20)
        print(f"Total jobs: {jobs.total}")
        print(f"Showing: {len(jobs.jobs)}")

        for job in jobs.jobs:
            print(f"  {job.job_id}: {job.status} | {len(job.urls)} URLs | Created: {job.created_at}")

        # Filter by status
        print("\n=== Completed Jobs ===")
        completed = await crawler.list_jobs(status="completed", limit=10)
        for job in completed.jobs:
            print(f"  {job.job_id}: {job.urls[0] if job.urls else 'N/A'}")

        print("\n=== Running Jobs ===")
        running = await crawler.list_jobs(status="running")
        print(f"Found {len(running.jobs)} running jobs")

        # Pagination example
        print("\n=== Pagination (Next 20) ===")
        page2 = await crawler.list_jobs(limit=20, offset=20)
        print(f"Page 2: {len(page2.jobs)} jobs")

        # Available statuses: pending, running, completed, failed, cancelled
        print("\n=== Failed Jobs ===")
        failed = await crawler.list_jobs(status="failed", limit=5)
        for job in failed.jobs:
            print(f"  {job.job_id}: {job.error or 'Unknown error'}")


if __name__ == "__main__":
    asyncio.run(main())

File: nodejs/examples/jobs/01_list_jobs_sdk.ts
#!/usr/bin/env npx ts-node
/**
 * Example: List jobs using SDK
 *
 * This example demonstrates:
 * - Listing all jobs with pagination
 * - Filtering jobs by status
 * - Accessing job metadata (timestamps, URLs, status)
 *
 * Usage:
 *     npx ts-node 01_list_jobs_sdk.ts
 */

import { AsyncWebCrawler } from 'crawl4ai-cloud';

// Configuration
const API_KEY = 'YOUR_API_KEY'; // Replace with your API key

async function main(): Promise<void> {
  const crawler = new AsyncWebCrawler({ apiKey: API_KEY });

  try {
    // List all jobs (default: 20 per page)
    console.log('=== All Jobs (First 20) ===');
    const jobs = await crawler.listJobs({ limit: 20 });
    console.log(`Total jobs: ${jobs.length}`);

    jobs.forEach((job) => {
      console.log(`  ${job.id}: ${job.status} | ${job.urlsCount} URLs | Created: ${job.createdAt}`);
    });

    // Filter by status
    console.log('\n=== Completed Jobs ===');
    const completed = await crawler.listJobs({ status: 'completed', limit: 10 });
    completed.forEach((job) => {
      console.log(`  ${job.id}`);
    });

    console.log('\n=== Running Jobs ===');
    const running = await crawler.listJobs({ status: 'running' });
    console.log(`Found ${running.length} running jobs`);

    // Pagination example
    console.log('\n=== Pagination (Next 20) ===');
    const page2 = await crawler.listJobs({ limit: 20, offset: 20 });
    console.log(`Page 2: ${page2.length} jobs`);

    // Available statuses: pending, running, completed, failed, cancelled
    console.log('\n=== Failed Jobs ===');
    const failed = await crawler.listJobs({ status: 'failed', limit: 5 });
    failed.forEach((job) => {
      console.log(`  ${job.id}: ${job.error || 'Unknown error'}`);
    });
  } finally {
    await crawler.close();
  }
}

main().catch(console.error);
